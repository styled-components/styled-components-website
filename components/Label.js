import styled from 'styled-components';
import rem from '../utils/rem';
import { blmGrey } from '../utils/colors';
import { headerFont } from '../utils/fonts';

export const LabelGroup = styled.div`
  display: inline-block;
  margin-left: 0.5rem;
  position: relative;
  bottom: ${rem(3)};
`;

const Label = styled.small`
  display: inline-block;
  background: ${(p) => (p.isVersion ? 'cornflowerblue' : blmGrey)};
  color: white;
  font-size: 0.75rem;
  font-family: ${headerFont};
  border-radius: ${rem(3)};
  padding: ${rem(1)} ${rem(5)};
  vertical-align: middle;
`;

export default Label;
