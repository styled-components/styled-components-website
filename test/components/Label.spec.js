import React from 'react';
import renderer from 'react-test-renderer';
import Label, { LabelGroup } from '../../components/Label';

test('Label renders correctly', () => {
  const tree = renderer.create(<Label />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('LabelGroup renders correctly', () => {
  const tree = renderer.create(<LabelGroup />).toJSON();

  expect(tree).toMatchSnapshot();
});
