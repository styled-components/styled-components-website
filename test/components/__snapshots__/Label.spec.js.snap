// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
.c0 {
  display: inline-block;
  background: rgb(219,112,147);
  color: white;
  font-size: 0.75rem;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  border-radius: 0.16666666666666666rem;
  padding: 0.05555555555555555rem 0.2777777777777778rem;
  margin-left: 0.4rem;
}

<small
  className="c0"
/>
`;

exports[`LabelGroup renders correctly 1`] = `
.c0 {
  display: inline-block;
  margin-left: 0.5rem;
  position: relative;
  bottom: 0.16666666666666666rem;
}

<div
  className="c0"
/>
`;
